cmd/private-prow-configs-mirror/main.go

```diff
@@ -0,0 +1,460 @@
+		repos := sets.NewString(lgtm.Repos...)
+
+		for _, orgRepo := range lgtm.Repos {
+			if orgs.hasOrgRepoFull(orgRepo) {
+				repo := strings.Split(orgRepo, "/")[1]
+
+				logger.WithField("repo", repo).Info("Found")
+				repos.Insert(privateOrgRepo(repo))
+			}
+		}
+
+		lgtms[index].Repos = repos.List()
+	}
+}
+
+func injectPrivateBugzillaPlugin(bugzillaPlugins plugins.Bugzilla, orgs orgsWithOfficialImages) {
+	logger := logrus.WithField("component", "plugins.bugzilla")
+	logger.Info("Processing...")
+
+	privateRepos := make(map[string]plugins.BugzillaRepoOptions)
+	for org, orgValue := range bugzillaPlugins.Orgs {
+		for repo, value := range orgValue.Repos {
+			if orgs.hasOrgRepo(org, repo) {
+				logger.WithField("repo", repo).Info("Found")
+				privateRepos[repo] = value
+			}
+		}
+	}
+
+	if len(privateRepos) > 0 {
+		logger.WithField("value", asYAML(privateRepos)).Info("Generating openshift-priv org.")
+		bugzillaPlugins.Orgs[openshiftPrivOrg] = plugins.BugzillaOrgOptions{Repos: privateRepos}
+	}
+}
+
+func injectPrivatePlugins(plugins map[string][]string, orgs orgsWithOfficialImages) {
+	logger := logrus.WithField("component", "plugins.plugins")
+	logger.Info("Processing...")
+
+	var allPrivate []string
+
+	for orgRepo, value := range plugins {
+		if orgs.hasOrgRepoFull(orgRepo) {
+			newValue := value
+
+			// We want to append the org level values, because the private repo
+			// can exist in different org with different configuration.
+			org := strings.Split(orgRepo, "/")[0]
+			if _, ok := plugins[org]; ok {
+				newValue = append(newValue, plugins[org]...)
+			}
+
+			allPrivate = append(allPrivate, newValue...)
+			plugins[privateOrgRepo(strings.Split(orgRepo, "/")[1])] = newValue
+		}
+	}
+
+	mergeCommonPrivatePlugins(plugins, getRepeatedValues(allPrivate), logger)
+}
+
+// mergeCommonPrivatePlugins detects the common values from all the private repositories
+// and merges them into the org `openshift-priv` level.
+// In addition, it generates a repo. if it contains an extra non-common field.
+func mergeCommonPrivatePlugins(plugins map[string][]string, repeatedValues sets.String, logger *logrus.Entry) {
+	privateOrgPlugins := sets.NewString()
+
+	for repo, values := range plugins {
+		if strings.HasPrefix(repo, openshiftPrivOrg) {
+			repoLevelPlugins := sets.NewString()
+
+			for _, p := range values {
+				if repeatedValues.Has(p) {
+					privateOrgPlugins.Insert(p)
+					continue
+				}
+				repoLevelPlugins.Insert(p)
+			}
+
+			if len(repoLevelPlugins.List()) == 0 {
+				delete(plugins, repo)
+			} else {
+				plugins[repo] = repoLevelPlugins.List()
+			}
+		}
+	}
+
+	if len(privateOrgPlugins.List()) > 0 {
+		logger.WithField("value", asYAML(privateOrgPlugins.List())).Info("Generating openshift-priv org.")
+		plugins[openshiftPrivOrg] = privateOrgPlugins.List()
+	}
+}
+
+func getRepeatedValues(values []string) sets.String {
+	temp := sets.NewString()
+	repeated := sets.NewString()
+
+	for _, value := range values {
```

Comment:
This should use set intersections

Reply:
Can't use set intersections here.  My purpose here is to grab a set of values that exist more than once in the `allPrivate []string`